### ISSUE TITLE
Support for max_instances argument

### ISSUE THREAD

**Issue by fpoon** [terraform-google-modules/terraform-google-scheduled-function#66]
### TL;DR

Hi,
It seems that this module lacks support for cloud function max_instances argument, though the underlying [event-function module seems to support it](https://registry.terraform.io/modules/terraform-google-modules/event-function/google/latest?tab=inputs#:~:text=Default%3A%20%7B%7D-,max_instances,-number).
Adding this would protect users against unlimited scaling in case of pubsub malfunction (e.g. constant failing to acknowledge message and redelivering it, which might result in increasing number of cloud function instances)

### Detailed design

```markdown
main.tf:
...
module "main" {
  ...
  max_instances = var.function_max_instances
  ...
}
...

variables.tf:
...
variable "function_max_intances" {
  type        = number
  default     = 0
  description = "The maximum number of parallel executions of the function."
}
...
```

### Additional information

If you are ok with this proposal, I'll be happy to contribute code :)

**Comment by bharathkkb** [comment]
@fpoon happy to review a PR adding this

**Comment by apeabody** [comment]
Completed in: https://github.com/terraform-google-modules/terraform-google-scheduled-function/pull/67
